orchestrator/nodes/ftp/index.js

```diff
@@ -0,0 +1,158 @@
+"use strict";
+
+var path = require('path');
+var logger = require('../../logger').logger;
+
+var util = require("util");
+
+var dojot = require('@dojot/flow-node');
+
+var stream = require("stream");
+var ftp = require("basic-ftp");
+
+class ReadStream extends stream.Readable {
+    constructor(object) {
+        super();
+        stream.Readable.call(this, {});
+        this._object = object;
+    }
+    _read() {
+        this.push(this._object);
+        this._object = null;
+    }
+}
+
+class DataHandler extends dojot.DataHandlerBase {
+    constructor() {
+        super();
+    }
+
+    /**
+     * Returns full path to html file
+     * @return {string} String with the path to the node representation file
+     */
+    getNodeRepresentationPath() {
+        return path.resolve(__dirname, 'ftp.html');
+    }
+
+    /**
+     * Returns node metadata information
+     * This may be used by orchestrator as a liveliness check
+     * @return {object} Metadata object
+     */
+    getMetadata() {
+        return {
+            'id': 'dojot/ftp',
+            'name': 'ftp',
+            'module': 'dojot',
+            'version': '1.0.0',
+        };
+    }
+
+        /**
+     * Returns full path to locales
+     * @returns {String} Path segments into an absolute path.
+     */
+    getLocalesPath() {
+        return path.resolve(__dirname, './locales');
+    }
+
+    /**
+     * Check if the node configuration is valid
+     * @param {object} config  Configuration data for the node
+     * @return {[boolean, object]} Boolean variable stating if the configuration is valid or not and error message
+     */
+    checkConfig() {
+
+        return [true, null];
+    }
+
+    /**
+     * Statelessly handle a single given message, using given node configuration parameters
+     *
+     * This method should perform all computation required by the node, transforming its inputs
+     * into outputs. When such processing is done, the node should issue a call to the provided
+     * callback, notifying either failure to process the message with given config, or the set
+     * of transformed messages to be sent to the flow's next hop.
+     *filename.txt
+     * @param  {[type]}       config   Node configuration to be used for this message
+     * @param  {[type]}       message  Message to be processed
+     * @return {[undefined]}
+     */
+    async handleMessage(config, message) {
+        logger.debug("Executing ftp node...");
+        logger.debug(`Config is: ${util.inspect(config)}`);
+        var url = config.url;
+        var tokens = config.url.match(/(ftp|ftps):\/\/(.*)/);
+        var transport = "ftp";
+        var host;
+        var port = 21;
+        var remaining;
+        var encoding = config.fileencoding;
+        const user = config.username;
+        const password = config.password; // THIS SHOULD NOT BE LIKE THIS!
```

Comment:
hum? What do you mean with this comment?

Reply:
It should be stored elsewhere in a safe way.

Reply:
Waiting for Vault to be included.

Reply:
agree, we must talk about it in the future.